types/globals.d.ts

```diff
@@ -1128,6 +1128,14 @@ interface Response extends Body {
   readonly statusText: string;
   // readonly type: ResponseType;
   readonly url: string;
+  /**
+   * Fastly-specific property - obtain the IP address used for the request
+   */
+  readonly ip: string;
+  /**
+   * Fastly-specific property - Obtain the port used for the request
+   */
+  readonly port: number;
```

Comment:
Non-blocking comment:

As these are only valid for responses which have come from a request, we could create a new TypeScript Interface for that scenario and update our `fetch` type definition to return it, that way it will be clear in the types that these properties are not on synthetic responses - what do you think?

```ts
interface FetchedResponse extends Response {
  /**
   * Fastly-specific property - obtain the IP address used for the request
   */
  readonly ip: string;
  /**
   * Fastly-specific property - Obtain the port used for the request
   */
  readonly port: number;
}


declare function fetch(input: RequestInfo | URL, init?: RequestInit): Promise<FetchedResponse>;
```

Reply:
It is also possible for these to not be set when the response comes from a request but that request is cached. Instead of providing an error I've just made both cases result in an `undefined` getter which I think better aligns with user expectations in that these values should typically be checked for truthiness before reading them in JS.

I think this design works well, and wiring through custom Response flavours would likely be unwieldy.